### DISCUSSION TITLE
Orientation random sampling

### DISCUSSION THREAD

**Discussion by xshang93** [mtex-toolbox/mtex#1417] [Ask Anything] [answered]
Hi, I'm using orientation.rand() to generate random orientations. Could anyone point out how the random orientations are generated, i.e., is it sampling from a normal distribution? I tried to plot Phi1, Phi and Phi2 but only Phi seems to follow a semi-normal distribution.
![phi1](https://user-images.githubusercontent.com/86078353/166067657-2aa0a5bc-d593-47e3-ab4f-cc8f74e60bb6.png)
![Phi](https://user-images.githubusercontent.com/86078353/166067668-bab149f6-5876-447f-95be-ab96f5688d8e.png)
![Phi2](https://user-images.githubusercontent.com/86078353/166067677-e6ae90b6-3012-4623-ad0e-ff8c47f1db09.png)

Thank you!

**Comment by ralfHielscher** [comment, marked as answer]
Hi xshang93,

could you please point out what you are doing? In my case 
``` matlab
cs = crystalSymmetry('432');
ori = orientation.rand(10000,cs)
[phi1,Phi,phi2] = Euler(ori)
histogram(phi1)
```
leads to an uniform histogram in the angles `phi1` and `phi2` as it should be. For the angle `Phi` the histogram is approximately the cos function. This is the correct definition of uniformly distributed orientations.

Ralf.

**Reply by xshang93** [reply]
Hi Ralf,

Thank you very much for your reply. You are definitely right. I just realized that for some reason I squared the angles, which is why it's not showing the uniform distribution.

Best,
Xiao
